internal/commands/app/create_inputs.go

```diff
@@ -131,6 +135,65 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
+func (i *createInputs) resolveTemplateID(ui terminal.UI, client realm.Client, templateID string) (string, error) {
```

Comment:
worth adding any unit tests here?  just to verify the various errors/prompts that can occur here

Reply:
unit tests for what specifically? We definitely don't want to directly unit test `resolveTemplateID` since it's a private method, but I can likely update my existing test cases to try to hit some of these additional branches.

Reply:
wait what's this "dont unit test private methods" idea.... have I been doing code wrong??

I'm thinking tests, somewhere, that at least prove the errors we return (any `Templates` error, the unable to find template error, the template not found error) as well a test that verifies our behavior when `--template` is not set but `--yes` is, behavior when `--template` is not set and no templates are found, and that the prompt eventually shows up and lets us pick an option

Reply:
I'm happy to test that behavior through the execution of the main commands itself, but testing private methods is pretty much always a bad idea unless there's a very specific reason for doing so. If someone changes the name of a private function that you're testing or makes some other internal refactoring change that doesn't actually alter the way the command behaves from the user's perspective, it's much more likely that that test will need to be updated to reflect the "new" version of the private behavior. It's a recipe for brittle tests.

Reply:
This whole codebase is brittle-y tested then... :bushes:

The idea I had in mind while writing tests for this CLI was: separate out inputs handling and command functionality.  I would still argue there's great value in unit testing our private, inputs-related functions: mostly around the various ways we intend to resolve said inputs.  in effect, it allowed us to, in detail, test the interactivity of CLI prompts and collecting inputs separately from the actual `Handler` implementation.  You could definitely just write these tests as `Handler` tests, but in my opinion that result is a bit overkill.

either way, it would be good to have something here that proves out the various errors we're returning, and the instances we actual prompt for more info

Reply:
yeah, I'll always prefer to see those tests at the `Handler` level since there doesn't seem to be much (anything?) you _can't_ test from that level, and it avoids the issue of coupling your tests to implementation details vs. end user behavior. I don't enjoy "fixing" tests for things that aren't actually broken.

Reply:
a lot of the input resolution in commands will happen before `Handler` is ever called, not really the case here with `resolveTemplateID`.  but lets at least have tests that flush out the various behaviors of `resolveTemplateID`, either with a test directly on that function (which I really still find to be fine here), or as variants of a tests on the `Handler` method.  basically the error cases and the interactive prompt case (and for the interactive test, you'll want to use a different form of the "mock ui", of which there are plenty of examples of)